operator/v1/types_ingress.go

```diff
@@ -585,6 +591,50 @@ type EndpointPublishingStrategy struct {
 	NodePort *NodePortStrategy `json:"nodePort,omitempty"`
 }
 
+// ClientCertificatePolicy describes the policy for client certificates.
+// +kubebuilder:validation:Enum=Required;Optional
+type ClientCertificatePolicy string
+
+const (
+	// ClientCertificatePolicyRequired indicates that a client certificate
+	// should be required.
+	ClientCertificatePolicyRequired ClientCertificatePolicy = "Required"
+
+	// ClientCertificatePolicyOptional indicates that a client certificate
+	// should be requested but not required.
+	ClientCertificatePolicyOptional ClientCertificatePolicy = "Optional"
+)
+
+// ClientTLS specifies TLS configuration to enable client-to-server
+// authentication, which can be used for mutual TLS.
+type ClientTLS struct {
+	// clientCertificatePolicy specifies whether the ingress controller
+	// requires clients to provide certificates.  This field accepts the
+	// values "Required" or "Optional".
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ClientCertificatePolicy ClientCertificatePolicy `json:"clientCertificatePolicy"`
+
+	// clientCA is a reference to a configmap containing the PEM-encoded CA
+	// certificate bundle that should be used to verify a client's
+	// certificate.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	ClientCA configv1.ConfigMapNameReference `json:"clientCA"`
+
+	// allowedSubjectPatterns specifies a list of regular expressions that
+	// should be matched against the distinguished name on a valid client
+	// certificate to filter requests.  If this list is empty, no filtering
+	// is performed.  If the list is nonempty, then at least one pattern
+	// must match a client certificate's distinguished name or else the
+	// ingress controller rejects the certificate and denies the connection.
```

Comment:
Is "regular expressions" specific enough, or is this field expecting a certain kind of regex (e.g. RE2).

Reply:
Good point, that really needs to be more specific.  We compile HAProxy with PCRE, so I've added a note that the regular expressions must use PCRE syntax.  Thanks!